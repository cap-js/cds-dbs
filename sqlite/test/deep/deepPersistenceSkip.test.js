const cds = require('../../../test/cds')
const { POST, PUT, DELETE } = cds.test(__dirname, 'deep.cds')

describe('deep operations with @cds.persistence.skip', () => {
  test('skip child to one with @cds.persistence.skip on deep insert', async () => {
    const uuid = cds.utils.uuid()
    const res = await POST('/bla/RootUUID', {
      ID: uuid,
      toOneSkip: {
        text: 'abc',
      },
    })
    expect(res.status).toBe(201)

    expect(res.data).toEqual({
      '@odata.context': '$metadata#RootUUID(toOneSkip())/$entity',
      ID: uuid,
      name: null,
      toOneChild_ID: null,
      toOneSkip_ID: expect.any(String),
    })
    expect(res.data.toOneSkip_ID).toBeDefined()
  })

  test('skip child to many with @cds.persistence.skip on deep insert', async () => {
    const uuid = cds.utils.uuid()
    const res = await POST('/bla/RootUUID', {
      ID: uuid,
      toManySkip: [{ text: 'a' }, { text: 'b' }],
    })
    expect(res.status).toBe(201)

    expect(res.data).toEqual({
      '@odata.context': '$metadata#RootUUID(toManySkip())/$entity',
      ID: uuid,
      name: null,
      toOneChild_ID: null,
      toOneSkip_ID: null,
    })
  })

  test('skip nested child to one with @cds.persistence.skip on deep insert', async () => {
    const uuid = cds.utils.uuid()
    const res = await POST('/bla/RootUUID', {
      ID: uuid,
      toOneChild: {
        text: 'abc',
        toManySubChild: [
          { text: 'a', toOneSkipChild: { text: 'aa' } },
          { text: 'b', toOneSkipChild: { text: 'bb' } },
        ],
      },
    })
    expect(res.status).toBe(201)

    expect(res.data).toEqual({
      '@odata.context': '$metadata#RootUUID(toOneChild(toManySubChild(toOneSkipChild())))/$entity',
      ID: uuid,
      name: null,
      toOneChild: {
        ID: expect.any(String),
        text: 'abc',
        toManySubChild: [
          {
            ID: expect.any(String),
            backlink_ID: res.data.toOneChild.ID,
            text: 'a',
            toOneSkipChild_ID: expect.any(String),
          },
          {
            ID: expect.any(String),
            backlink_ID: res.data.toOneChild.ID,
            text: 'b',
            toOneSkipChild_ID: expect.any(String),
          },
        ],
      },
      toOneChild_ID: res.data.toOneChild.ID,
      toOneSkip_ID: null,
    })
  })

  test('skip child to one with @cds.persistence.skip on deep update', async () => {
    const uuid = cds.utils.uuid()
    const resPost = await POST('/bla/RootUUID', {
      ID: uuid,
      toOneSkip: {
        text: 'abc',
      },
    })
    expect(resPost.status).toBe(201)

    const resUpdate = await PUT(`/bla/RootUUID(${uuid})`, {
      name: 'abc',
      toOneSkip: {
        text: 'cd',
      },
    })
    expect(resUpdate.status).toBe(200)

    expect(resUpdate.data).toEqual({
      '@odata.context': '$metadata#RootUUID/$entity',
      ID: uuid,
      name: 'abc',
      toOneChild_ID: null,
      toOneSkip_ID: expect.any(String),
    })
  })

  test('skip child to one with @cds.persistence.skip on deep delete', async () => {
    const uuid = cds.utils.uuid()
    const resPost = await POST('/bla/RootUUID', {
      ID: uuid,
      toOneSkip: {
        text: 'abc',
      },
    })
    expect(resPost.status).toBe(201)

    const resUpdate = await DELETE(`/bla/RootUUID(${uuid})`)
    expect(resUpdate.status).toBe(204)
  })
})
