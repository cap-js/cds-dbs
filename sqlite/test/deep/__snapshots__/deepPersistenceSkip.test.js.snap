// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[` 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[` 3`] = `"prepare - sL/lh1Iw6g6ufy8G1sveuh7e1gs="`;

exports[` 4`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 5`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 6`] = `"prepare - rFK0VW86dBPY+WLEZhIZG+ZKADQ="`;

exports[` 7`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 8`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 9`] = `"prepare - yRMXewqXMJIix1Bv5Z0XbW82YTk="`;

exports[` 10`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 11`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 12`] = `"prepare - hxRK6XUbeTzdtB1GCyt1//INBlU="`;

exports[` 13`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 14`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 15`] = `"prepare - ZU0RvTJ6nw0z5WL3r/mrTG65Bsg="`;

exports[` 16`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 17`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 18`] = `"prepare - s2ayCCJQkeyDCFvY2LNOPGhsSlw="`;

exports[` 19`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 20`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 21`] = `"prepare - /Fo7oRi2bcsk2wWetaEgc67b7wE="`;

exports[` 22`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 23`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 24`] = `"prepare - +l/U6cwRmqeX6sal8OHcZbKePYI="`;

exports[` 25`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 26`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 27`] = `"prepare - /TLN6uippqbCWB54Q8+QuPkccgg="`;

exports[` 28`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 29`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 30`] = `"prepare - lHdhkw9AEjbSNAyuCt5xxLrNGX0="`;

exports[` 31`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 32`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 33`] = `"prepare - RrnMv4SmaFlIRH5v8emVDPt0iNE="`;

exports[` 34`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 35`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 36`] = `"prepare - J/gor6IGC7jqdgOcZS3SgPWldxU="`;

exports[` 37`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 38`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 39`] = `"prepare - HDVNVS6vMtVtvv1XgtuZVLrRuyU="`;

exports[` 40`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 41`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 42`] = `"prepare - Y3osuMdrqkmWOB5ywO6zjOp43rg="`;

exports[` 43`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 44`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 45`] = `"prepare - lRoxrLxTxTAC40iYX2rJ5ffX6t0="`;

exports[` 46`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 47`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 48`] = `"prepare - TFZ+vlo9KzrEHeAczRXYv225698="`;

exports[` 49`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 50`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 51`] = `"prepare - +349rwEolW9qFanScpu6TmpVtWQ="`;

exports[` 52`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 53`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 54`] = `"prepare - HQGOhPU3b2tw+FyLdiW5ZXOJPCs="`;

exports[` 55`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 56`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 57`] = `"prepare - kHALwOJDdMV4v+EsUFKeKv11DBk="`;

exports[` 58`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 59`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 60`] = `"prepare - GXRIb/sL73trzCp2RnUXty07J3w="`;

exports[` 61`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 62`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 63`] = `"prepare - C8RnfYuWGCpm/lh4nKa6bLz1adE="`;

exports[` 64`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 65`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 66`] = `"prepare - ANWdtBRY3R5xlWktH3N+j1bAeZI="`;

exports[` 67`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 68`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 69`] = `"prepare - iCzGf7Q+LDFThWk/TQfPjOrp8bk="`;

exports[` 70`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 71`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 72`] = `"prepare - xC4wWrKhuueWbP+OmzSX5En1amg="`;

exports[` 73`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 74`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 75`] = `"prepare - ZxoczIWlTYRPU3ujGX+rZRM+HHE="`;

exports[` 76`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 77`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 78`] = `"prepare - PQqg+qJrGj+Bi6kStoF8HiIFWyQ="`;

exports[` 79`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 80`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 81`] = `"prepare - JzNT6q+ZbuCLDCZGOOo+zNJ5P5o="`;

exports[` 82`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 83`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 84`] = `"prepare - yAnqOV7M4iYz+OnZ3LiYrKRZ6bw="`;

exports[` 85`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 86`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 87`] = `"prepare - mZa44kKv2XM/zc/6gKOdp4N/+7c="`;

exports[` 88`] = `"run - UUdcj0oDV17Ffi3tQ6LAx9SMd7Q="`;

exports[` 89`] = `"{"changes":0,"lastInsertRowid":0}"`;

exports[` 90`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[` 91`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 3`] = `"prepare - uUQ/f4z7BQeTqf5OXzLrEPc1YvE="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 4`] = `"run - cd8fF5qig2Cj0JDMgW2QlQ2zHSA="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 5`] = `"{"changes":1,"lastInsertRowid":2}"`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 6`] = `"prepare - EuOBmh748GnifcFT4S/p64cwwZg="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 7`] = `"all - 9xJhJ2iyL8OvO3FYOdxyTLSP5T8="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 8`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000003\\",\\"name\\":null,\\"toOneChild_ID\\":null,\\"toOneSkip_ID\\":null}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 9`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to many with @cds.persistence.skip on deep insert 10`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 3`] = `"prepare - uUQ/f4z7BQeTqf5OXzLrEPc1YvE="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 4`] = `"run - q/TQAJsGc3IzaCw/9DL8xE7iUbM="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 5`] = `"{"changes":1,"lastInsertRowid":5}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 6`] = `"prepare - EuOBmh748GnifcFT4S/p64cwwZg="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 7`] = `"all - IfKRO3gy1KKQju8PKpw35dL4p/c="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 8`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000019\\",\\"name\\":null,\\"toOneChild_ID\\":null,\\"toOneSkip_ID\\":\\"00000000-0000-0000-0000-000000000021\\"}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 9`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 10`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 11`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 12`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 13`] = `"prepare - eorwrE9JeUPTZjXEEwrUtYnt1n4="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 14`] = `"all - IfKRO3gy1KKQju8PKpw35dL4p/c="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 15`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000019\\",\\"toOneChild\\":null}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 16`] = `"prepare - sabY8KdoKtvuQDhITaDq01zWwjI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 17`] = `"run - IfKRO3gy1KKQju8PKpw35dL4p/c="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 18`] = `"{"changes":1,"lastInsertRowid":5}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 19`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep delete 20`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 3`] = `"prepare - uUQ/f4z7BQeTqf5OXzLrEPc1YvE="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 4`] = `"run - 5VhyRkBcwaAqmfXyJcUlTnI+JS4="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 5`] = `"{"changes":1,"lastInsertRowid":1}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 6`] = `"prepare - EuOBmh748GnifcFT4S/p64cwwZg="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 7`] = `"all - wQuUa52XAjRm6SQqPZ9epoVAqrs="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 8`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000000\\",\\"name\\":null,\\"toOneChild_ID\\":null,\\"toOneSkip_ID\\":\\"00000000-0000-0000-0000-000000000002\\"}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 9`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep insert 10`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 3`] = `"prepare - uUQ/f4z7BQeTqf5OXzLrEPc1YvE="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 4`] = `"run - cOJJBPfnQiFWNHZ/0WdYTOiq8GE="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 5`] = `"{"changes":1,"lastInsertRowid":4}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 6`] = `"prepare - EuOBmh748GnifcFT4S/p64cwwZg="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 7`] = `"all - 0JSsnpnKpT27kI1HM2WFxVI3+Qo="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 8`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000014\\",\\"name\\":null,\\"toOneChild_ID\\":null,\\"toOneSkip_ID\\":\\"00000000-0000-0000-0000-000000000016\\"}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 9`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 10`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 11`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 12`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 13`] = `"prepare - StZw/aIMIJME5OtEEn38WnD18a8="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 14`] = `"all - 0JSsnpnKpT27kI1HM2WFxVI3+Qo="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 15`] = `"[{"_json_":"{\\"name\\":null,\\"ID\\":\\"00000000-0000-0000-0000-000000000014\\",\\"toOneSkip_ID\\":\\"00000000-0000-0000-0000-000000000016\\",\\"toOneChild_ID\\":null}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 16`] = `"prepare - aexuup6TxH/+rtSornOA5W4gQXo="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 17`] = `"run - X5NgZ7p8ZTfW6ZqJCHW2UGelkGc="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 18`] = `"{"changes":1,"lastInsertRowid":4}"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 19`] = `"prepare - EuOBmh748GnifcFT4S/p64cwwZg="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 20`] = `"all - 0JSsnpnKpT27kI1HM2WFxVI3+Qo="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 21`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000014\\",\\"name\\":\\"abc\\",\\"toOneChild_ID\\":null,\\"toOneSkip_ID\\":\\"00000000-0000-0000-0000-000000000018\\"}"}]"`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 22`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip child to one with @cds.persistence.skip on deep update 23`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 1`] = `"exec - swFX3mSaTpBcIO4VaMo4btiduC8="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 2`] = `"{"name":":memory:","open":true,"inTransaction":true,"readonly":false,"memory":true}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 3`] = `"prepare - uUQ/f4z7BQeTqf5OXzLrEPc1YvE="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 4`] = `"run - 8CptpsjcAWgY0wGFYFhr8RiJXRc="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 5`] = `"{"changes":1,"lastInsertRowid":3}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 6`] = `"prepare - XJ+uDE9fLcriQE0CdxR86PdeG7w="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 7`] = `"run - 8rRc5VXFNSMU3e2NAfKsoPPLV08="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 8`] = `"{"changes":1,"lastInsertRowid":1}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 9`] = `"prepare - iD5Z7jcPKQljRtcAohmVSVJrVp8="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 10`] = `"run - an+oaKJOZdf5IlsaqrF4HPpgtpM="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 11`] = `"{"changes":1,"lastInsertRowid":1}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 12`] = `"prepare - iD5Z7jcPKQljRtcAohmVSVJrVp8="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 13`] = `"run - XXunnrFbDVoCwouQUDBQn0QCw4w="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 14`] = `"{"changes":1,"lastInsertRowid":2}"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 15`] = `"prepare - ksL6IHgBDdQ7gFXEBq16/UEur38="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 16`] = `"all - 7SQfMjFH3sAyM4Ce2LW/ukXGC4E="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 17`] = `"[{"_json_":"{\\"ID\\":\\"00000000-0000-0000-0000-000000000007\\",\\"name\\":null,\\"toOneChild\\":{\\"ID\\":\\"00000000-0000-0000-0000-000000000009\\",\\"text\\":\\"abc\\",\\"toManySubChild\\":[{\\"ID\\":\\"00000000-0000-0000-0000-000000000010\\",\\"text\\":\\"a\\",\\"backlink_ID\\":\\"00000000-0000-0000-0000-000000000009\\",\\"toOneSkipChild_ID\\":\\"00000000-0000-0000-0000-000000000011\\"},{\\"ID\\":\\"00000000-0000-0000-0000-000000000012\\",\\"text\\":\\"b\\",\\"backlink_ID\\":\\"00000000-0000-0000-0000-000000000009\\",\\"toOneSkipChild_ID\\":\\"00000000-0000-0000-0000-000000000013\\"}]},\\"toOneChild_ID\\":\\"00000000-0000-0000-0000-000000000009\\",\\"toOneSkip_ID\\":null}"}]"`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 18`] = `"exec - mvfei0i6/P7jaJW/2LTxQWnJXeI="`;

exports[`deep operations with @cds.persistence.skip skip nested child to one with @cds.persistence.skip on deep insert 19`] = `"{"name":":memory:","open":true,"inTransaction":false,"readonly":false,"memory":true}"`;
